lib/puppet/provider/x509_cert/openssl.rb

```diff
@@ -99,7 +99,13 @@ def create
         '-out', resource[:path]
       ]
     end
-    options << ['-passin', "pass:#{resource[:password]}"] if resource[:password]
+    if resource[:cakey_password]
```

Comment:
I'm worried about the missing else case because then `password` is undefined. I'd prefer `password = resource[:cakey_password] || resource[:password]`

Reply:
So that it ends up as nil explicitly rather impicitly?